subsys/bluetooth/shell/l2cap.c

```diff
@@ -275,6 +275,80 @@ static int cmd_register(const struct shell *sh, size_t argc, char *argv[])
 	return 0;
 }
 
+#if defined(CONFIG_BT_L2CAP_ECRED)
+static int cmd_ecred_reconfigure(const struct shell *sh, size_t argc, char *argv[])
+{
+	struct bt_l2cap_chan *l2cap_ecred_chans[1] = {&l2ch_chan.ch.chan};
+	uint16_t mtu;
+	int err;
+
+	if (!default_conn) {
+		shell_error(sh, "Not connected");
+		return -ENOEXEC;
+	}
+
+	if (!l2ch_chan.ch.chan.conn) {
+		shell_error(sh, "Channel not connected");
+		return -ENOEXEC;
+	}
+
+	mtu = strtoul(argv[1], NULL, 10);
+
+	err = bt_l2cap_ecred_chan_reconfigure(l2cap_ecred_chans, mtu);
+	if (err < 0) {
+		shell_error(sh, "Unable to reconfigure channel (err %d)", err);
+	} else {
+		shell_print(sh, "L2CAP reconfiguration pending");
+	}
+
+	return err;
+}
+
+static int cmd_ecred_connect(const struct shell *sh, size_t argc, char *argv[])
+{
+	struct bt_l2cap_chan *l2cap_ecred_chans[1] = {&l2ch_chan.ch.chan};
+	uint16_t psm;
+	int err;
+
+	if (!default_conn) {
+		shell_error(sh, "Not connected");
+
+		return -ENOEXEC;
+	}
+
+	if (l2ch_chan.ch.chan.conn) {
+		shell_error(sh, "Channel already in use");
+
+		return -ENOEXEC;
+	}
+
+	psm = shell_strtoul(argv[1], 16, &err);
+	if (err) {
```

Comment:
`shell_strtoul` doesn't override `err` on success, so we need to initialize `err = 0` before we can do this check (not quite sure why it was implemented that way...)